jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/auth/AWSCredentialProvider.java

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright 2018 T Mobile, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License.  You may obtain a copy
+ * of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ ******************************************************************************/
+package com.tmobile.pacbot.azure.inventory.auth;
+
+import org.springframework.stereotype.Component;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.BasicSessionCredentials;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import com.amazonaws.services.securitytoken.model.AssumeRoleRequest;
+import com.amazonaws.services.securitytoken.model.AssumeRoleResult;
+
+
+/**
+ * The Class CredentialProvider.
+ */
+@Component
+public class AWSCredentialProvider {
+	
+	/** The dev mode. */
+	private static boolean devMode = System.getProperty("PIC_DEV_MODE")==null?false:true;
+	
+	
+	/**
+	 * Gets the base account credentials.
+	 *
+	 * @param roleName the role name
+	 * @return the base account credentials
+	 */
+	public BasicSessionCredentials getCredentials (String baseAccount, String baseRegion,String roleName){
+		if(devMode){
+			String accessKey = System.getProperty("ACCESS_KEY"); 
+			String secretKey = System.getProperty("SECRET_KEY"); 
+			BasicAWSCredentials awsCreds = new BasicAWSCredentials(accessKey, secretKey);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

If possible, use temporary security credentials (IAM roles) instead of long-term access keys.
Long-term access keys, such as those associated with IAM users and AWS account root users, remain valid until you manually revoke them. However, temporary security credentials obtained through IAM roles and other features of the AWS Security Token Service expire after a short period of time. Use temporary security credentials to help reduce your risk in case credentials are accidentally exposed.
Learn more about [best practices for managing AWS access keys](https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html).